### DISCUSSION TITLE
Addition in List Collection

### DISCUSSION THREAD

**Discussion by anantupadhyay5786** [nhibernate/nhibernate-core#2774] [Q&A]
Hi,

We are trying to use list collection in our project.
Sample code used in .hbm is as follows for Department.hbm.xml :

 <list name="emplList" lazy="extra" cascade="all">
      <key column="DEPARTMENT_ID" not-null="true"/>
      <index column="POSITION" />
      <one-to-many class="Employee"/>
    </list> 

For employee.hbm.xml :

<many-to-one name="emplDepartment" class="Department" not-null="true" insert="false" update="false" cascade="all"
            column="DEPARTMENT_ID"/>

The issue is during Add of a record in list, eager loading is happening on list collection in nHibernate code due to below code:
protected bool IsOperationQueueEnabled
		{
			get { return !initialized && IsConnectedToSession && IsInverseCollection; }
		}

 indexed collection will not use inverse="true", therefore for us this flag is false. Hence eager load. Therefore , We want to know, the role of IsInverseCollection flag in function IsOperationQueueEnabled. Moreover , We are taking care of inverse from many-to-one side as well. Is it still needed for List collection to be loaded in-memory before any add operation ?

**Comment by fredericDelaporte** [comment]
It sound to me there is some misunderstanding about `inverse` here. If it is a one-to-many, and if you handle the association update from the many-to-one side, then you should map that list as being the `inverse` end, solving by the way your laziness trouble. An `inverse` end is the side of the association which is not responsible for updating the association. So if you also nullify the other end when having something to remove, you should map the list as `inverse`.
